specification/cdn/resource-manager/Microsoft.Cdn/stable/2020-09-01/afdx.json

```diff
@@ -2165,15 +2165,6 @@
             "required": true,
             "type": "string"
           },
-          {
-            "name": "ruleSet",
```

Comment:
>"name": "ruleSet" [](start = 12, length = 17)

Is this intentional? a put call without a body? where are the properties being set now?

Reply:
Yes. There are no real properties on the rule set. It's just a container / parent for the rules.

Reply:
What's the api adding the rules?

---
In reply to: [569876409](https://github.com/Azure/azure-rest-api-specs/pull/12798#discussion_r569876409) [](ancestors = 569876409)

Reply:
It's right below it and it adds it to the rulesets, which looks like profiles/rulesets/rules

Reply:
Just having a placeholder doesn't seem ok, what happens if the endpoint /rulesets/rules gets invoked first instead of this placeholder? Is better to model your endpoint in such a way that one call creates your rule set and inner rules. @chiragg4u for more insight.

Reply:
@jorgecotillo Ruleset is a container that represents a set of rules that can be associated to Route (another resource). The significance of the ruleset is that it makes it easy to for the user to associate a set of rules to multiple routes, which is a very common use case for Frontdoor users. 

Please note that even though ruleSet doesn't have any properties, its a valid container resource. If the endpoint ruleSet/rules gets invoked first, it will fail with 404 as the parent ruleset resource doesn't exist. 

 Rules can be added/modified and removed from a ruleSet independently and so we anyways need a CRUD on individual rules, and thus the put on ruleset with inner rules feel redundant. Besides, each Ruleset may currently have upto 200 Rules, so asking to add/modify them all in a single https call introduces other complications like Https Content-Length overflow.

Please let me know if you have further concerns.

Reply:
@ShirleyJ27 please find the RPC requirement @ https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#put-resource 
Since you've confirmed in the checklist looks like you've missed this part from the documentation.
Please note there are many other products assumes ARM Resource structure and generically process the resources. In case you don't follow any of the common property many such scenario will break.

In this specific case it will also break the contract and will be impacting the SDK's. 

Can you provide the reasoning on why you are trying to remove this?

Reply:
I'm not sure which part of it breaks the contract, unless a payload is required for PUT calls.
The reasoning behind removing the PUT calls is because there are not actual properties on rulesets. Having it there would just require an empty payload from the customer, in the form of an empty bracket ("{}"), and that seems like a bad customer experience.

Reply:
You've only changed the object in PUT but you are still returning the object in GET; not sure how will you be able to return the object.

Reply:
We are returning the provisioning state so that signifies whether the call went through. It doesn't have other info on it because the object doesn't have any "real" attributes on it.

Reply:
in your manifest you have logging rules setup on this type to log the request and response body. I don't know if that will break if you remove the requirement for a request body. So unless there is a S360 kpi requirement that requires this change I would recommend not making this change, and have the user specify an empty content "{}" explicitly

---
In reply to: [571176681](https://github.com/Azure/azure-rest-api-specs/pull/12798#discussion_r571176681) [](ancestors = 571176681)